src/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs

```diff
@@ -28,48 +28,86 @@
 namespace Microsoft.Identity.Client.AppConfig
 {
     /// <summary>
-    ///     Options object with string values loadable from JSON configuration (as in an asp.net configuration scenario)
+    /// Options object with string values loadable from a configuration file 
+    /// (for instance a JSon file as in an asp.net configuration scenario)
+    /// See https://aka.ms/msal-net-application-configuration
     /// </summary>
     public abstract class ApplicationOptions
     {
         /// <summary>
+        /// Client ID (also known as App ID) of the application as registered in the
+        /// application registration portal (https://aka.ms/msal-net-register-app)
         /// </summary>
         public string ClientId { get; set; }
 
         /// <summary>
-        /// Can be domain, can be guid tenant, can be meta-tenant (e.g. consumers).
+        /// Tenant from which the application will allow users to sign it. This can be:
+        /// a domain associated with a tenant, a guid (tenant id), or a meta-tenant (e.g. consumers).
+        /// This property is mutually exclusive with <see cref="AadAuthorityAudience"/>
         /// </summary>
+        /// <remarks>The name of the property was chosen o ensure compatibility with AzureAdOptions 
+        /// in ASP.NET Core configuration files (even the semantics would be tenant)</remarks>
         public string TenantId { get; set; }
 
         /// <summary>
-        /// Mutually exclusive with TenantId...
+        /// Sign-in audience. This property is mutually exclusive with TenantId...
         /// </summary>
         public AadAuthorityAudience AadAuthorityAudience { get; set; } = AadAuthorityAudience.None;
 
         /// <summary>
-        /// For compat with AzureAdOptions...
+        /// STS instance (for instance https://login.microsoftonline.com for the Azure public cloud). 
+        /// The name was chosen to ensure compatibility with AzureAdOptions in ASP.NET Core.
+        /// This property is mutually exclusive with <see cref="AzureCloudInstance"/>
```

Comment:
"Instance" is a very overloaded word, I propose we find something better. How about we name these things as the enum + "Override" suffix, i.e. AzureCloudInstanceOverride ? It will also make it clear which one overrides which.